### PR TITLE
Improve Input component

### PR DESCRIPTION
**Pull request by Rheniery**
If applied, pull request will improve the Input component to receive label, message, and description.

### Other minor changes:
 - Categories were removed from the Input component.

### Card Link:
N/A

### Design Expected Screenshot
N/A

### Implementation Screenshot or GIF
Seleção_117

### Notes:
N/A

### CODE REVIEW

File: src/components/Input/DefaultInput/DefaultInput.css [hunk 1]
```diff
@@ -44,3 +99,38 @@
...
+.label {
+  font-size: var(--input-label-font-size);
+  font-weight: var(--input-label-font-weight);
+  line-height: var(--input-label-line-height);
+  color: var(--input-label-color);
+  display: flex;
+  width: 100%;
+}
+
+.span {
```
Comment (kaiomagalhaes) [inline src/components/Input/DefaultInput/DefaultInput.css:112]: please avoid creating classes without a context. A `span` doesn't mean anything other than an HTML tag. A better naming could be 'description'

File: src/components/Input/DefaultInput/index.tsx [hunk 2]
```diff
@@ -1,38 +1,76 @@
...
+    placeholder,
+    value,
+    variablesClassName,
+    size,
+    state,
+    ...inputProps
+  } = props;
   const sizeClass = `input--${size}`;
+  const messageClass = state ? `input--message--${state}` : '';
+  const validateClass = state ? `input--${state}` : '';
```
Comment (kaiomagalhaes) [inline src/components/Input/DefaultInput/index.tsx:36]: this should be a `statusClass`

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: src/components/Input/index.tsx [hunk 3]
```diff
@@ -1,26 +1,26 @@
 import React from 'react';
 
-import SimpleInput from '@/components/Input/SimpleInput';
+import DefaultInput from '@/components/Input/DefaultInput';
```
Comment (diogoribeiro) [inline src/components/Input/index.tsx:3]: We can merge the DefaultInput and the Input component. They're the same component, so it's not necessary to hide the input's implementation in another component

File: src/components/Input/index.tsx [hunk 4]
```diff
@@ -1,26 +1,88 @@
...
+        className={classNames(
+          props.className,
+          styles.input,
+          styles[sizeClass],
+          styles[statusClass],
+          variablesClassName
+        )}
+        onChange={onChange}
+        placeholder={placeholder}
+        type='text'
```
Comment (diogoribeiro) [inline src/components/Input/index.tsx:63]: let's remove this prop. This prop can be handled by the `inputProps`

File: src/components/Input/index.tsx [hunk 5]
```diff
@@ -1,26 +1,88 @@
...
-import SimpleInput from '@/components/Input/SimpleInput';
+import styles from '@/components/Input/Input.css';
 
-type Category = 'simple';
 type Size = 'large' | 'medium';
+type State = '' | 'valid' | 'invalid';
 
 interface Props {
-  category?: Category;
+  className: string;
```
Comment (diogoribeiro) [inline src/components/Input/index.tsx:10]: this component should allow only the `variablesClassName`. What's the idea of having the `className` prop?

PS.: to modify the input look & feel, the dev needs to use the `variablesClassName` and pass the new values for the variables

**Review by diogoribeiro — CHANGES_REQUESTED** [review]

File: src/components/Input/Input.stories.tsx [hunk 6]
```diff
@@ -9,9 +9,8 @@ export default {
 
 const Template = args => <Input {...args} />;
 
-export const Simple = Template.bind({});
-Simple.args = {
-  category: 'simple',
+export const Default = Template.bind({});
```
Comment (kaiomagalhaes) [inline src/components/Input/Input.stories.tsx:12]: please add a default, positive, and a negative example

File: src/components/Input/index.tsx [hunk 7]
```diff
@@ -1,26 +1,80 @@
...
+        ''
+      )}
+    </div>
+  );
 };
 
-const Input: React.FC<Props> = props => {
-  const InputType = inputs[props.category || 'simple'];
-  return <InputType {...props} />;
+Input.defaultProps = {
```
Comment (kaiomagalhaes) [inline src/components/Input/index.tsx:70]: we don't need to add empty values for the optional attributes.

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: src/components/Input/Input.stories.tsx [hunk 8]
```diff
@@ -9,9 +9,23 @@ export default {
...
+Default.args = {
   size: 'large',
-  placeholder: 'Enter text'
+  placeholder: 'Enter text',
+  state: ''
```
Comment (kaiomagalhaes) [inline src/components/Input/Input.stories.tsx:16]: do you need to pass this attribute?

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: src/components/Input/Input.stories.tsx (hunk 8 above)
Comment (Rheniery) [reply src/components/Input/Input.stories.tsx:16]: No. I don't, it's just an example

**Review by diogoribeiro — APPROVED** [review]

**Pull request merged**